### PR TITLE
Set Loading State for Entities

### PR DESCRIPTION
**Pull request by googlvalenzuela**
## Summary
Declare executing state of an entity based on verb states.
This allows Android Auto scenarios to show that the entity is loading and provid
`
fun <T> Entity<T>.isExecuting()=  state.endsWith("ing")
`

## Screenshots
Screenshot 2023-04-29 at 11 07 35 AM
e feedback for the user about the state on which the entity is in.

### CODE REVIEW

**Comment by marazmarci** [comment]
> States is an open system and can be any value. While I think this works for the currently supported domains, to prevent showing the 'loading' indicator unintentionally IMO the `isExecuting` function should check against a list of known executing states instead to be safe.

a list of known states can be found here I think: https://github.com/home-assistant/core/blob/ef9bcd9285980385eede726e15aa1e7d20238eee/homeassistant/const.py#L330

**Review by JBassett — CHANGES_REQUESTED** [review]
I agree, just checking if the state ends in ing is too broad.

**Comment by googlvalenzuela** [comment]
Thanks for the feedback, Ive updated the code.

File: common/src/main/java/io/homeassistant/companion/android/common/data/integration/Entity.kt [hunk 1]
```diff
@@ -642,3 +642,13 @@ fun <T> Entity<T>.friendlyState(context: Context): String {
...
     return friendlyState
 }
+
+fun <T> Entity<T>.isExecuting() = when (state) {
+    "closing" -> true
+    "locking" -> true
+    "opening" -> true
+    "unlocking" -> true
+    "buffering" -> true
+    "disarming" -> true
```
Comment (jpelgrom) [inline common/src/main/java/io/homeassistant/companion/android/common/data/integration/Entity.kt:652]: If you include `disarming`, also add `arming`

**Comment by googlvalenzuela** [comment]
Added Armig.

**Review by jpelgrom — APPROVED** [review]

**Review by JBassett — APPROVED** [review]

**Pull request merged**